docs/program-setup/bare_metal/baremetal_usage.md

```diff
@@ -0,0 +1,75 @@
+# Using bare metal profile
```

Comment:
This requires an intro that explains why you need a specific workflow, and what this page covers.

Reply:
```suggestion
# Using bare metal profile

This guide shows how to create a bare metal profile application: 

1. By default, when you build an application binary, the build tool uses the full profile. To use the bare metal profile, you need to set up your application to override this default behaviour. 
1. Bare metal has a minimal set of APIs. You can add additional ones [from the list of supported API]().
1. Bare metal has its own set of tests in Greentea that can verify <!--what does it verify? just the default APIs?-->
```

Reply:
> what does it verify? just the default APIs?

It's a really good question. _As far as I understand_, we initially wanted to verify the default APIs only. But some of the tests couldn't compile, so we brought in more APIs just to fix the existing compilation issue. The side effect? More APIs are tested! (https://github.com/ARMmbed/mbed-os/pull/11721)

With this explanation, I don't think there is a clear logic why we verify some APIs and not others with the bare metal profile, but that's just my personally understanding. I will discuss this with the team.

Reply:
Okay, it's simply all tests that currently support the bare metal profile (and the target, obviously)